src/protocol/control_layer/ControlMessageFactory.js

```diff
@@ -0,0 +1,68 @@
+import UnsupportedVersionError from '../../errors/UnsupportedVersionError'
+import PublishRequest from './publish_request/PublishRequest'
+import PublishRequestFactory from './publish_request/PublishRequestFactory'
+import SubscribeRequest from './subscribe_request/SubscribeRequest'
+import SubscribeRequestFactory from './subscribe_request/SubscribeRequestFactory'
+import UnsubscribeRequest from './unsubscribe_request/UnsubscribeRequest'
+import UnsubscribeRequestFactory from './unsubscribe_request/UnsubscribeRequestFactory'
+import BroadcastMessage from './broadcast_message/BroadcastMessage'
+import BroadcastMessageFactory from './broadcast_message/BroadcastMessageFactory'
+import UnicastMessage from './unicast_message/UnicastMessage'
+import UnicastMessageFactory from './unicast_message/UnicastMessageFactory'
+import SubscribeResponse from './subscribe_response/SubscribeResponse'
+import SubscribeResponseFactory from './subscribe_response/SubscribeResponseFactory'
+import UnsubscribeResponse from './unsubscribe_response/UnsubscribeResponse'
+import UnsubscribeResponseFactory from './unsubscribe_response/UnsubscribeResponseFactory'
+import ResendResponseResending from './resend_response_resending/ResendResponseResending'
+import ResendResponseResendingFactory from './resend_response_resending/ResendResponseResendingFactory'
+import ResendResponseResent from './resend_response_resent/ResendResponseResent'
+import ResendResponseResentFactory from './resend_response_resent/ResendResponseResentFactory'
+import ResendResponseNoResend from './resend_response_no_resend/ResendResponseNoResend'
+import ResendResponseNoResendFactory from './resend_response_no_resend/ResendResponseNoResendFactory'
+import ResendLastRequestV1 from './resend_request/ResendLastRequestV1'
```

Comment:
Referring to our discussion: the pattern where messages statically register themselves as handlers to a `(type, version)` tuple would IMO help resolve this "import hell", make things easier to maintain (don't need to touch this class when new protocol versions are introduced), and contribute to better separation of concerns (individual classes are the only ones who need to know what they do, and declare themselves).